invenio_app_ils/records/api.py

```diff
@@ -65,7 +91,176 @@ def update(self, data):
         super(IlsRecord, self).update(data)
 
 
-class Document(IlsRecord):
+class IlsRecordWithRelations(IlsRecord):
+    """Ils records class with relations."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize ILS record with relations."""
+        super(IlsRecordWithRelations, self).__init__(*args, **kwargs)
+        self.updated_related_records = []
+
+    @staticmethod
+    def edition_relation():
+        """Get edition relation type."""
+        return current_app.config["EDITION_RELATION"]
+
+    @staticmethod
+    def language_relation():
+        """Get language relation type."""
+        return current_app.config["LANGUAGE_RELATION"]
+
+    @staticmethod
+    def get_relation_by_id(relation_id):
+        """Get the relation_type by id."""
+        for relation in current_app.config["PIDRELATIONS_RELATION_TYPES"]:
+            if relation.id == relation_id:
+                return relation
+        raise RelatedRecordError(
+            "No relation type with id: {}".format(relation_id)
+        )
+
+    @property
+    def editions(self):
+        """Get related editions."""
+        return self._get_related(self.edition_relation())
+
+    @property
+    def languages(self):
+        """Get related languages."""
+        return self._get_related(self.language_relation())
+
+    def node(self, relation_type):
+        """Get PIDNode for this record."""
+        return PIDNode(pid=self.pid, relation_type=relation_type)
+
+    def add_related_edition(self, record):
+        """Add record as a related edition record."""
+        self.add_related(record, self.edition_relation())
+
+    def remove_related_edition(self, record):
+        """Remove related edition record."""
+        self.remove_related(record, self.edition_relation())
+
+    def add_related_language(self, record):
+        """Add record as a related language record."""
+        self.add_related(record, self.language_relation())
+
+    def remove_related_language(self, record):
+        """Remove related language record."""
+        self.remove_related(record, self.language_relation())
+
+    def add_related(self, record, relation_type):
+        """Add related record with a relation."""
+        self._add_related_record(
+            record,
+            relation_type,
+            parent_node=self._parent_node(relation_type)
+        )
+
+    def remove_related(self, record, relation_type):
+        """Remove related record with a relation."""
+        self._remove_related_record(
+            record,
+            relation_type,
+            parent_node=self._parent_node(relation_type)
+        )
+
+    def _add_related_record(self, record, relation_type, parent_node=None):
+        """Add related record with a relation."""
+        record_node = record.node(relation_type)
```

Comment:
if the passed `record` is not of type `IlsRecordWithRelation`, here it will fail right? Does it make sense maybe raise before?

Reply:
Yes, I think we should raise when doing validation. Initially I was gonna add the validation in this PR but it seemed easier to include it in the PR for the UI so unless you prefer to have it now I'll add it in the backoffice PR.